src/rgw/rgw_asio_frontend.cc

```diff
@@ -424,13 +427,30 @@ int AsioFrontend::init()
       lderr(ctx()) << "failed to open socket: " << ec.message() << dendl;
       return -ec.value();
     }
+
+    if (l.endpoint.protocol() == tcp::v6()) {
+      l.acceptor.set_option(boost::asio::ip::v6_only(true), ec);
+      if (ec) {
+        lderr(ctx()) << "failed to set v6_only socket option: "
+		     << ec.message() << dendl;
+	return -ec.value();
+      }
+    }
+
     l.acceptor.set_option(tcp::acceptor::reuse_address(true));
     l.acceptor.bind(l.endpoint, ec);
     if (ec) {
+      if (ec == boost::asio::error::operation_not_supported) {
```

Comment:
the error appears to be `boost::asio::error::address_family_not_supported`, and it comes from `acceptor.open()` rather than `acceptor.bind()`. tested by booting with `ipv6.disabled=1`